plugin/validator_test.go

```diff
@@ -1561,6 +1569,114 @@ func TestValidator_ValidateAllTypedefs(t *testing.T) {
 			wantErr:     true,
 			expectedErr: `{"config":{"sni":"must not be an IP"}}`,
 		},
+		{
+			name: "valid certificate",
+			config: jsonify(map[string]string{
+				"certificate": pluginTesting.Cert,
+			}),
+		},
+		{
+			name: "valid ecdsa-with-SHA256 certificate",
+			config: jsonify(map[string]string{
+				"certificate": pluginTesting.CertEcdsa,
+			}),
+		},
+		{
+			name: "valid webclient certificate",
+			config: jsonify(map[string]string{
+				"certificate": pluginTesting.CertClient,
+			}),
+		},
+		{
+			name: "valid webclient2 certificate",
+			config: jsonify(map[string]string{
+				"certificate": pluginTesting.CertClient2,
+			}),
+		},
+		{
+			name: "expired certificate",
+			config: jsonify(map[string]string{
+				"certificate": pluginTesting.CertAlt,
+			}),
+		},
+		{
+			name: "valid alt ecdsa certificate",
+			config: jsonify(map[string]string{
+				"certificate": pluginTesting.CertAltEcdsa,
+			}),
+		},
+		{
+			name: "expired alt-alt certificate",
+			config: jsonify(map[string]string{
+				"certificate": pluginTesting.CertAltAlt,
+			}),
+		},
+		{
+			name: "valid alt-alt ecdsa certificate",
+			config: jsonify(map[string]string{
+				"certificate": pluginTesting.CertAltAltEcdsa,
+			}),
+		},
+		{
+			name: "valid CA certificate",
+			config: jsonify(map[string]string{
+				"certificate": pluginTesting.CertCA,
+			}),
+		},
+		{
+			name: "valid key",
+			config: jsonify(map[string]string{
+				"key": pluginTesting.Key,
+			}),
+		},
+		{
+			name: "valid ecdsa-with-SHA256 key",
+			config: jsonify(map[string]string{
+				"key": pluginTesting.KeyEcdsa,
+			}),
+		},
+		{
+			name: "valid webclient key",
+			config: jsonify(map[string]string{
+				"key": pluginTesting.KeyClient,
+			}),
+		},
+		{
+			name: "valid webclient2 key",
+			config: jsonify(map[string]string{
+				"key": pluginTesting.KeyClient2,
+			}),
+		},
+		{
+			name: "valid alt key",
+			config: jsonify(map[string]string{
+				"key": pluginTesting.KeyAlt,
+			}),
+		},
+		{
+			name: "valid alt ecdsa key",
+			config: jsonify(map[string]string{
+				"key": pluginTesting.KeyAltEcdsa,
+			}),
+		},
+		{
+			name: "valid alt-alt key",
+			config: jsonify(map[string]string{
+				"key": pluginTesting.KeyAltAlt,
+			}),
+		},
+		{
+			name: "valid alt-alt ecdsa key",
+			config: jsonify(map[string]string{
+				"key": pluginTesting.KeyAltAltEcdsa,
+			}),
+		},
+		{
+			name: "valid CA key",
+			config: jsonify(map[string]string{
+				"key": pluginTesting.KeyCA,
+			}),
+		},
```

Comment:
Can you please add 2 tests with invalid key and cert format?

```
		{
			name: "invalid key",
			config: jsonify(map[string]string{
				"key": "bla",
			}),
		},
		{
			name: "invalid cert",
			config: jsonify(map[string]string{
				"certificate": "bla",
			}),
		},
```

Reply:
yep!